src/sage/manifolds/differentiable/vectorfield_module.py

```diff
@@ -950,8 +952,13 @@ def alternating_contravariant_tensor(self, degree, name=None,
                                       latex_name=latex_name)
         return self.exterior_power(degree).element_class(self, degree,
                                        name=name, latex_name=latex_name)
+    @overload
+    def alternating_form(
+        self, degree: Literal[0], name=None, latex_name=None
+    ) -> ScalarField:
+        pass
 
-    def alternating_form(self, degree, name=None, latex_name=None):
+    def alternating_form(self, degree: int, name=None, latex_name=None) -> DiffForm:
```

Comment:
``int`` is not correct. It also accepts Sage `Integer` and many other things that can be coerced to ``int``.

Reply:
It's not completely correct. To correctly type things using the coercion framework is https://github.com/sagemath/sage/issues/30511.